File: Books/Competitive_Programming_3/Chapter_1/Section_133/Normal/UVa_00573/UVa_00573.java
package Books.Competitive_Programming_3.Chapter_1.Section_133.Normal.UVa_00573;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class UVa_00573 {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s;

        while ((s = br.readLine()) != null) {
            StringTokenizer st = new StringTokenizer(s);
            int h = Integer.parseInt(st.nextToken());
            if (h == 0) break;
            double u = Integer.parseInt(st.nextToken());
            double d = Integer.parseInt(st.nextToken());
            double f = Integer.parseInt(st.nextToken());
            double curr = 0;
            int day = 0;
            double ff = u * f/100;
            while (true) {
                day++;
                if (u > 0) curr += u;
                u = u - ff;
                if (curr > h) break;
                curr = curr - d;
                if (curr < 0) break;
            }
            if (curr < 0) System.out.println(new StringBuilder().append("failure on day ").append(day).toString());
            else System.out.println(new StringBuilder().append("success on day ").append(day).toString());
        }
    }
}

File: Books/Competitive_Programming_4/Chapter_1/Section_4/Medium/UVa_Judge/UVa_00573/UVa_00573.c
#include <stdio.h>

int main()
{
    while ( 1 )
    {
        int h, d, f;
        float u;
        scanf("%d %f %d %d", &h, &u, &d, &f);
        if ( h == 0 ) break;
        float c = 0.0f;
        int day = 0;
        float ff = 0.1f;
        ff = u * f / 100;
        while ( 1 )
        {
            day++;
            if ( u > 0 ) c += u;
            u = u - ff;
            if ( c > h ) break;
            c = c - d;
            if ( c < 0 ) break;
        }
        if ( c < 0 ) printf("failure on day %d\n", day);
        else printf("success on day %d\n", day);
    }
}
